### PR TITLE
FindImageOverlaps test

### PR DESCRIPTION
**Pull request by acpaquette**
Adds a new gtest for the findimageoverlaps application.

## Description
Adds a new gtest for the findimageoverlaps function that adds footprint polygons to a set of cubes, applies the findimageoverlaps function and checks the return values.

## Related Issue
None (Remake of an old PR that was clobbered #3836)

## Motivation and Context
This is another application that needs to be converted to a suite of gtests at some point.

## How Has This Been Tested?
This is a test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have added myself to the [AUTHORS.rst](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/AUTHORS.rst) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by Kelvinrr — COMMENTED** [review]
Looks pretty clean, the only thing is make sure the old Make test is deleted for default tests. 

Also, this is still missing the "full overlap" test from the original Make tests so we can't check this off completely yet. If you don't think you have time to get to that today, I'm cool with merging and taking care of it since I'm not sure I have enough for today.

**Review by Kelvinrr — CHANGES_REQUESTED** [review]
Fixtures.cpp seems to have build errors on CentOS.

File: isis/tests/FuntionalTestsFindImageOverlaps.cpp [hunk 1]
```diff
@@ -50,4 +67,104 @@ TEST_F(ThreeImageNetwork, FunctionalTestFindImageOverlapsNoOverlap) {
...
+  int polyStrSize = polyStr.size();
+  std::istringstream polyStream(polyStr);
+
+  Blob pvlBlob("Footprint", "Polygon");
+  Pvl pvl;
+  PvlObject polyObject = PvlObject("Polygon");
+  polyObject.addKeyword(PvlKeyword("Name", "Footprint"));
+  polyObject.addKeyword(PvlKeyword("StartByte", "1"));
+  polyObject.addKeyword(PvlKeyword("Bytes", toString(polyStrSize)));
+  pvl.addObject(polyObject);
```
Comment (Kelvinrr) [inline isis/tests/FuntionalTestsFindImageOverlaps.cpp:135]: I have a passing thought that putting this boilerplate into a function is a good idea, but with the new fixture I'm not sure we will be doing this super often going forward anyways.

**Review by Kelvinrr — DISMISSED** [review]
Think this is good to go after CI finishes.

**Comment by jlaura** [comment]
@acpaquette Looks like this needs a rebase? Because of the stats work.

File: isis/tests/FuntionalTestsFindImageOverlaps.cpp (hunk 1 above)
Comment (acpaquette) [reply isis/tests/FuntionalTestsFindImageOverlaps.cpp:135]: That's something I was considering but I don't know how worth while it is like you mention.

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**